app/domain/util/open_ssl/x509/certificate.rb

```diff
@@ -0,0 +1,86 @@
+# A declarative constructor for an X509::Certificate
+#
+# TODO: this could pulled out into a gem
+
+require 'openssl'
+require 'active_support/time'
+
+module Util
+  module OpenSsl
+    module X509
+      module Certificate
+
+        # Create any cert, specify any of the options
+        #
+        def self.from_hash(
+          subject:,
+          issuer:,
+          public_key:,
+          good_for:, # accepts any object with to_i
```

Comment:
Would `expires_in` be a better name for this var? I guess they both get the point across, but `good_for` seems like it could be considered an English idiom.

Reply:
Yeah I considered that and this one might be a matter of taste.  I actually went with `good_for` because I sorta felt its plain talk, good ol american directness was less ambiguous than the more formal `expires_in` (which might be confused with `expires_at`)   But I'm not married to it by any means.

We could also change this when we gem-ify this code.

Reply:
Okay, I would still lean towards the more formal offering just because it's more in line with the vernacular an engineer would expect to see when dealing with such things, though I see your point that it's quite easy to mix up those two letters.